### ISSUE TITLE
simplify (?) doc/setup/postgresql.md acls handling

### ISSUE THREAD

**Issue by landryb** [georchestra/georchestra#1060]
First, we start by creating the georchestra database with the default ownership, and then grant all privs on it to the newly created georchestra db user - why not giving it ownership right away, using createuser *then* createdb -O ?

As for the schemas handling, so far in the doc for mapfishapp/ldapadmin/ogcstats/geofence we : 
- load the sql script as 'psql root' user, which creates the schema and tables
- fix/grant acls afterwards to the 'regular' db user, doing several grants as 'psql root' user.

this looks over-complicated. From the limited testing i've done, doing 'create schema foo authorization theuser;' is enough for 'theuser' to have permission to create tables/sequences in schema foo - and thus we might be able to run the script creating tables as 'theuser'. no need for the grant commands then...

same thing when we enable the postgis extension for geofence, iirc 'create extension postgis;' in recent debian versions (pgis 2.1, maybe 2.0, iirc postgis.control which enables 'create extension' is available since 2.0) is enough for 'regular db users without specific ACLs on spatial_ref_sys and {geography,geometry}_columns' to be able to import data into postgis with shp2psql - this has to be rechecked of course. It shouldnt also benecessary to run postgis.sql and spatial_ref_sys.sql manually...

To sum it up, i'd separate the steps/commands to run between : 
- what *needs* to be done as 'psql root user' : create user, db, schemas, extension
- what can be done as regular user with enough acls: create tables and insert records

**Comment by landryb** [comment]
In fact, schema creation can even happen as the user that owns the database - only extension creation has to be done as superuser, and afterwards everything can be done as regular user.
```
$sudo -u postgres createdb -E utf8 -T template0 -O georchestra test
$psql -h localhost -U georchestra -d test -c 'create schema testschema;'
CREATE SCHEMA
$psql -h localhost -U georchestra -d test -c 'create extension postgis;'
ERROR:  permission denied to create extension "postgis"
HINT:  Must be superuser to create this extension.
$sudo -u postgres psql -d test -c 'create extension postgis;'
CREATE EXTENSION
$shp2pgsql /srv/data/geonetwork3/index/spatialindex.shp testschema.spatialindex | psql -h localhost -d test -U georchestra
```

And the georchestra user, as owner of the database, doesnt need special rights to import geospatial data in postgis, including writing to geometry_columns and such (shp2pgsql does 'SELECT AddGeometryColumn' at the beginning)

this is with debian packages postgresql-9.4 '9.4.4-1.pgdg80+1' and postgis '2.1.7+dfsg-3~94.git954a8d0.pgdg80+1' from apt.postgresql.org - but should be the same with packages from jessie or jessie-backports.

**Comment by landryb** [comment]
```createlang plpgsql``` is not needed either, it seems in 9.4 it's loaded by default, at least as a dependency of postgis ?

**Comment by landryb** [comment]
```
$sudo -u postgres createdb -E utf8 -T template0 -O georchestra test2
$sudo -u postgres psql -d test2 -c '\dx'
                 List of installed extensions
  Name   | Version |   Schema   |         Description
---------+---------+------------+------------------------------
 plpgsql | 1.0     | pg_catalog | PL/pgSQL procedural language
```
plpgsql is even loaded by default in new databases.

**Comment by Vampouille** [comment]
If you use georchestra pg user, you will have to type password for each install step. I think it's better to use a sudo command that can be scripted.
About `create schema foo authorization`, there is only one file that contains references to user name "www-data" (current postgres user in georchestra doc). If we add authorization on CREATE SCHEMA then "www-data" user will appear in multiple files :

* mapfishapp/database.sql
* ldapadmin/database.sql
* ogc-server-statistics/databse.sql
* geofence/doc/setup/sql/002_create_schema_postgres.sql

I think it is better to keep references to database users in only one file.

You can look at :

https://github.com/georchestra/geofence/blob/georchestra-15.06/doc/setup/sql/002_create_schema_postgres.sql

user reference has been removed.

**Comment by Vampouille** [comment]
See https://github.com/georchestra/georchestra/pull/1099

**Comment by landryb** [comment]
Sorry but i still think it's awkward to run the insert/create scripts as psql root user and grant privileges afterwards :) Would be better to advise the user to create the .pgpass line with the correct perms so that he can access the db with the right user without having to type the pwd...

**Comment by fvanderbiest** [comment]
I see your point Landry, but I'm really not sure we'll have time to change the way it is currently done.
Please provide a PR for 15.12 if you think it's worth it.
